### DISCUSSION TITLE
Display Code Coverage in Pull Request

### DISCUSSION THREAD

**Discussion by NathanGrenier** [sonic-howl/frc2024#29] [General]
Do you know if it's possible to host html files generated by a workflow step?

There's a command that generated a summary of the test code coverage as html and it would be nice to be able to view that in the PR. 
I tried implementing this by:
1. Hosting the files (Done by the action called `html-preview-action@v2`)
2. Posting a link to the hosted files in a comment on the PR

As discussed in [this PR](https://github.com/sonic-howl/frc2024/pull/26), there seems to be a pre-made action to do just that but I couldn't get it to work.

**This isn't a huge priority** (not a big deal if it's impossible), just wanted to know if you've done something similar before.

**Comment by relishcolouredhat** [comment]
@NathanGrenier Can you link to the workflow that will create the html file?

**Reply by relishcolouredhat** [reply]
I'm having trouble finding the failed workflow run - unless the issue is "the task isn't being executed" 

I think I found the yml; and created a different repo to test in case 'not being executed' is the issue ;

with the same `integrate.yml` file; the `HTML Preview` task completes successfully ( see [link](https://github.com/relishcolouredhat/gha-testing/actions/runs/7634616503/job/20798762712) )

I believe it does fail to actually output when I follow the 'test' url's though -

The echo htmlpreview thing I haven't seen before, and doesn't seem to be working? 

I think we will need to place the output into a directory to view, this path seems to be looking for it in root of project but file does not exist

**Reply by NathanGrenier** [reply]
> I'm having trouble finding the failed workflow run - unless the issue is "the task isn't being executed"

The issue was that the task wasn't running properly (although it does run successfully)

> I believe it does fail to actually output when I follow the 'test' url's though -

This is correct. Id generates the hmtl for the coverage, but the action to host and link to those files isn't working.

> The echo htmlpreview thing I haven't seen before, and doesn't seem to be working?

This was for testing purposes. It should be removed if we get it to output a correct url

> I think we will need to place the output into a directory to view, this path seems to be looking for it in root of project but file does not exist

This is likely the case. Like I mentioned, I didn't have a lot of time to debug it (just got home from school now).

The issue is most likely due to either:
- The action looking in the wrong place
- The action is broken. 
  - It is mentioned in the action's GitHub (in an issue) that it doesn't work for generated files, only files already present in the repo. This is why I was asking you if you knew any alternatives.

Again, this isn't a high priority issue, so if you're not sure about an alternative, it's not a big deal.
